e2etest/pkg/frr/bfd.go

```diff
@@ -0,0 +1,23 @@
+// SPDX-License-Identifier:Apache-2.0
+
+package frr
+
+import (
+	"github.com/pkg/errors"
+	"go.universe.tf/metallb/e2etest/pkg/executor"
+	bgpfrr "go.universe.tf/metallb/internal/bgp/frr"
+)
+
+// BFDPeers returns informations for the all the bfd peers in the given
+// executor.
+func BFDPeers(exec executor.Executor) (map[string]bgpfrr.BFDPeer, error) {
+	res, err := exec.Exec("vtysh", "-c", "show bfd peers json")
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to query neighbours")
+	}
+	peers, err := bgpfrr.ParseBFDPeers(res)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to parse neighbours %s", res)
+	}
+	return peers, nil
+}
```

Comment:
so the CI as it stands it make sure BFD sessions are created , how about the failure detection part how does that been verified/tested, do we know how long it takes for BFD enabled session to detect a failure ? any plans to cover this in CI or at least manual testing for now and doc it ?

Reply:
Failover tests are in the backlog. We'd need to carefully think about them, because we are still subject to the time it takes ot execute the commands. Maybe setting a ludicrous hold time on the bgp side and to check that the failover happens in a lower time would work.
I'd keep this on the functional side though.